cmd/ris-mirror/rismirror/router.go

```diff
@@ -14,16 +15,17 @@ import (
 type Router struct {
 	name        string
 	address     net.IP
-	vrfRegistry *vrf.VRFRegistry
-	vrfs        map[uint64]*_vrf
+	vrfs        map[uint64]*vrfWithMergedLocRIBs // this is the authoritative data store for VRFs
+	vrfsMu      sync.RWMutex
+	vrfRegistry *vrf.VRFRegistry // this is only there so that the metrics functionality of the vrf package can be used
```

Comment:
I think the comment is not true, given the 

```
r.vrfRegistry.CreateVRFIfNotExists(vrf.RouteDistinguisherHumanReadable(rd), rd)
```

below :-D

Reply:
The point is when you remove/add VRFs you have to do it on the `vrfs` member first, holding the mutex. `vrfRegistry` is only there so that metrics of the VRFs can be exposed. It not authoritative.